hype-submitter/src/main/java/com/spotify/hype/runner/LocalDockerRunner.java

```diff
@@ -0,0 +1,188 @@
+        );
+      }
+
+      hostConfig.appendBinds(HostConfig.Bind.from(credentials)
+                               .to("/etc/gcloud/key.json")
+                               .readOnly(true)
+                               .build());
+
+      // Mount temporary file to act as the termination log
+      // Use user home because Docker Engine daemon has only limited access to on macOS or
+      // Windows filesystem
+      final Path localTmp = new File(System.getProperty("user.home")).toPath().resolve(".tmp");
+      final Path termLogs = Files.createDirectories(localTmp.resolve("spotify-hype-termination-logs"));
+      final Path terminationLog = Files.createTempFile(termLogs, "termination-log", ".txt");
+      if(!keepTerminationLog) {
+        terminationLog.toFile().deleteOnExit();
+      }
+      hostConfig.appendBinds(HostConfig.Bind
+          .from(terminationLog.toString())
+          .to("/dev/termination-log")
+          .readOnly(false)
+          .build());
+
+      hostConfig.appendBinds(HostConfig.Bind
+          .from(runSpec.stagedContinuation().manifestPath().getParent().toString())
+          .to(STAGING_VOLUME)
+          .readOnly(false)
+          .build());
+
+      Path volumes = Files.createDirectories(localTmp.resolve("spotify-hype-volumes"));
+      runSpec.runEnvironment().volumeMounts().forEach(m -> {
+        String localVolume = volumes.resolve(m.volumeRequest().id()).toString();
+        hostConfig.appendBinds(HostConfig.Bind
+            .from(localVolume)
+            .to(m.mountPath())
+            .build());
+      });
+
+      final File stagingContinuationFile = stagedContinuation.manifestPath().toFile();
+
+      final ContainerConfig containerConfig = ContainerConfig.builder()
+          .image(imageWithTag)
+          .cmd(of("file://" + STAGING_VOLUME + "/" + stagingContinuationFile.getName()))
+          .hostConfig(hostConfig.build())
+          .build();
+      creation = client.createContainer(containerConfig);
+      client.startContainer(creation.id());
+      LOG.info("Started container {}", creation.id());
+      final Optional<URI> uri = blockUntilComplete(creation.id(), terminationLog);
+      if(!keepContainer) {
+        client.removeContainer(creation.id());
+      }
+      return uri.map(u -> stagingContinuationFile.toPath()
+          .resolveSibling(Paths.get(u).toFile().getName()).toUri());
+    } catch (DockerException | IOException e) {
+      throw new RuntimeException("Failed to start docker container", e);
+    } catch (InterruptedException e) {
+      throw new RuntimeException("Interrupted while blocking", e);
+    }
+  }
+
+  private Optional<URI> blockUntilComplete(final String containerId,
+                                           final Path terminationLog) throws InterruptedException {
+    while (true) {
+      final ContainerInfo containerInfo;
+      try {
+        containerInfo = client.inspectContainer(containerId);
+
+        if (!containerInfo.state().running()) {
```

Comment:
Is there a race condition here? Could this method be checking the container before it starts running? Or does `client.startContainer()` block until container is running?